### DISCUSSION TITLE
MSBuild targets for publishing extensions

### DISCUSSION THREAD

**Discussion by reduckted** [VsixCommunity/Community.VisualStudio.Toolkit#256] [Ideas]
For one of my extensions I created an MSBuild target to publish the extension to the Visual Studio Marketplace. You can see it here:

https://github.com/reduckted/GitWebLinks/blob/3cb46cb99f2657d1e8e70c32de571df7a29e656b/visual-studio/build/PublishToMarketplace.targets

You can publish by running:

```
msbuild /t:PublishToMarketplace /p:Configuration=Release
```

I was planning on making this into a NuGet package so that I don't have to write it for every one of my extensions, but I think it would be cool if the toolkit provided a build target like this. 

Perhaps a build target to publish to the VSIX Gallery would be useful as well? Are there any other common tasks that we could write build targets for?

**Comment by ErikEJ** [comment]
Maybe burning the version number?

**Reply by reduckted** [reply]
That's a great idea!

I forgot that I actually wrote a target to do that for one of my extensions ([here](https://github.com/reduckted/ProjectFilter/blob/5efc9732a72d8402896c38f2690da3ec6311102a/source/ProjectFilter/ProjectFilter.csproj#L129-L146)). I do remember running into a bit of trouble with it though, because the code-behind that's generated from the VSIX Synchronizer extension isn't updated! Thankfully that extension didn't use the `Vsix.Version` property, but we'd definitely need to do make sure it's updated as well.

**Comment by madskristensen** [comment]
I'm using a GitHub Action to perform all those things now. I just finished it over the holidays. See it here: https://github.com/madskristensen/PkgdefLanguage/blob/master/.github/workflows/build.yaml

**Reply by ErikEJ** [reply]
Nice  - But we dont all use Github actions

**Reply by madskristensen** [reply]
I know, it was more for inspiration, I guess

**Reply by ErikEJ** [reply]
Appreciated!

**Reply by reduckted** [reply]
I used to use `cezarypiatek/VsixPublisherAction`, but ran into https://github.com/microsoft/VSExtensibility/issues/70 recently, which is why I ended up writing an MSBuild target.
